iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/TreeFactory.java

```diff
@@ -133,6 +135,10 @@ public KeyValuePairTree keyValuePairEquals(SyntaxToken key, SyntaxToken equals,
     return new KeyValuePairTreeImpl(key, equals, value);
   }
 
+  public UserTree user(SyntaxToken keyword, SyntaxToken user, Optional<SyntaxToken> colon, Optional<SyntaxToken> group) {
```

Comment:
WDYT about using a tuple here? Colon and group are always a pair. In the constructor, you can separate them again.

Reply:
Excellent suggestion ! That's what I was looking for, I was not satisfied with this.